templates/al2023/runtime/rootfs/etc/systemd/system/soci-snapshotter.service

```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=soci snapshotter containerd plugin
+Documentation=https://github.com/awslabs/soci-snapshotter
+
+[Service]
+Type=notify
+ExecStart=/usr/bin/soci-snapshotter-grpc --address fd://
```

Comment:
For my future self -- the `--address fd://` option here tells the snapshotter to look for file descriptors passed to the process by `systemd` as part of the socket activation flow: https://github.com/coreos/go-systemd/blob/7d375ecc2b092916968b5601f74cca28a8de45dd/activation/files_unix.go#L39

`systemd` will own the "listening" socket, and will create a "proxy" fd to route traffic to the backend service after it's started.